core/keyboard_nav/navigation.js

```diff
@@ -190,6 +138,34 @@ Blockly.navigation.focusFlyout_ = function() {
   }
 };
 
+/**
+ * Finds where the cursor should go on the workspace. This is either the top
+ * block or a set position on the workspace.
+ * @private
+ */
+Blockly.navigation.focusWorkspace_ = function() {
+  Blockly.hideChaff();
+  var cursor = Blockly.getMainWorkspace().getCursor();
+  var reset = Blockly.getMainWorkspace().getToolbox() ? true : false;
+  var topBlocks = Blockly.getMainWorkspace().getTopBlocks(true);
+
+  Blockly.navigation.resetFlyout_(reset);
+  Blockly.navigation.currentState_ = Blockly.navigation.STATE_WS;
+  if (topBlocks.length > 0) {
+    cursor.setCurNode(Blockly.navigation.getTopNode(topBlocks[0]));
+  } else {
+    var ws = Blockly.getMainWorkspace();
```

Comment:
This should be declared earlier so you can use it instead of calling Blockly.getMainWorkspace() every time in this function.

Reply:
done